frontend/src/metabase/dashboard/components/PublicLinkPopover/QuestionPublicLinkPopover.tsx

```diff
@@ -0,0 +1,61 @@
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import {
+  exportFormats,
+  publicQuestion as getPublicQuestionUrl,
+} from "metabase/lib/urls";
+import {
+  createPublicLink,
+  deletePublicLink,
+} from "metabase/query_builder/actions";
+import { getSetting } from "metabase/selectors/settings";
+import type Question from "metabase-lib/Question";
+import { PublicLinkPopover } from "./PublicLinkPopover";
+import type { ExportFormatType } from "./PublicLinkPopover";
+
+export const QuestionPublicLinkPopover = ({
+  question,
+  target,
+  isOpen,
+  onClose,
+}: {
+  question: Question;
+  target: JSX.Element;
+  isOpen: boolean;
+  onClose: () => void;
+}) => {
+  const dispatch = useDispatch();
+
+  const siteUrl = useSelector(state => getSetting(state, "site-url"));
```

Comment:
I have other comments about `siteUrl`. Basically I'm not sure what we try to achieve by passing it to `getPublicQuestionUrl` when the case that we don't pass it, we still get the settings from `MetabaseSettings.get("site-url")` anyway.